### DISCUSSION TITLE
How to use pinia orm with computed properties

### DISCUSSION THREAD

**Discussion by garma83** [CodeDredd/pinia-orm#957] [Q&A] [answered]
Hi,

I might be missing something obvious. One reason im using pinia orm is to make sure data is not recomputed when nothing changed. In VuexOrm or Vuex (or Pinia?) I would use Getters for this, which, as I understand, are basically computed references. However I can't quite get this to work properly with Pinia ORM. 

I have a model called Location, which needs to keep a currentId for the currently selected record. I added a static function to the model like this: 

```
export default class Location extends Model {

    static piniaOptions = {
        state: () => ({
            data: {},
            currentId: null,
        }),
    }
    static current():  {
        let currentId = useRepo(this).piniaStore().currentId
        return currentId ? useRepo(this).useCache(this.entity).find(currentId) : null;
    }
}

```

This function 'current()' I am using all over the place, in all kinds of components. It is not uncommon for this function to be called 10+ times during 1 page load. Within the components I call the function like this:

`const currentLocation = computed(() => Location.current());
`

What I see is that the query is ran for every component. How do I avoid this? Especially the hydration of models adds up over time

Maybe what I want is that the pinia orm store keeps the computed reference that I can then use?

I tried using the cache (in the current call, and then clearing it in a saving hook), but it doesnt play nice with computed references. when the cache is cleared, the computed property doesnt update.

**Comment by garma83** [comment, marked as answer]
I have something of a solution maybe? But its not working yet. Would appreciate feedback.

I have the below in the model:

    static piniaOptions = {
        state: () => ({
            data: {},
            currentId: <number | string | null>null,
        }),
        getters: {
            current(this: DataStore) {
                return this.currentId ? this.data[this.currentId] : null;
            }
        }
    }


The above doesnt work because it doesnt return a valid instance (pinia orm did not work its magic on it)

One complicating factor is that in the getter I dont have access to the Model directly. All models are inherited from a single model (Called MyModel) . This is so I dont have to repeat all the logic for currents (and a few other things). However because of this, this doesnt work in the getter:

useRepo(MyModel).find(this.currentId)

Ideally I would be able to use Location in the MyModel getter, but I can't figure out how to do that. The DataStore doesnt have a reference to Location, as far as I can see.

**Reply by CodeDredd** [reply]
Well i dont know yet if the solution i have first in mind is the best.
On your example you could use this:
````ts
static piniaOptions = {
    state: () => ({
        data: {},
        currentId: <number | string | null>null,
    }),
    getters: {
        current(this: DataStore) {
            return this.currentId ? useRepo(MyModel).revive(this.data[this.currentId]) : null;
        }
    }
}
````

This way you don't lose the caching. It would help me a lot if you would have a working example on codesandbox or stackblitz.
